include/oneapi/dpl/experimental/kt/esimd_radix_sort_cooperative.h

```diff
@@ -22,19 +22,24 @@
 namespace oneapi::dpl::experimental::esimd::impl
 {
 
+template <uint32_t STAGES>
 void inline init_global_sync(uint32_t * psync, uint32_t tg_id) {
     using namespace __ESIMD_NS;
     using namespace __ESIMD_ENS;
-    simd<uint32_t, 16> lane_id(0, 4);
-    simd<uint32_t, 16> old_value = lsc_atomic_update<atomic_op::load, uint32_t, 16>(psync, lane_id, 1);
+
+    constexpr uint32_t SYNC_BUFFER_PER_STAGE = 4;
+    constexpr uint32_t SYNC_BUFFER_SIZE = SYNC_BUFFER_PER_STAGE * STAGES;
```

Comment:
We should have here 4 items for each stage.
So initial const 16 is OK for 4 stages only.
For 8 stages we should have 32 instead 16.